### PR TITLE
add wifi scan example

### PR DESCRIPTION
**Pull request by kbleeke**
Adds an example for wifi scanning that just prints all the discovered networks

### CODE REVIEW

File: examples/rpi-pico-w/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#![no_std]
```
Comment (Dirbaio) [inline examples/rpi-pico-w/src/lib.rs:1]: i'd just copy-paste this into each example, even if that means there's a bit of duplication. "single crate with both lib and bin targets" can be confusing for Rust newcomers, it's nice that examples are as simple as possible as a self-contained `.rs` file.

Comment (kbleeke) [reply examples/rpi-pico-w/src/lib.rs]: true

File: examples/rpi-pico-w/src/bin/tcp_server.rs [hunk 2]
```diff
@@ -18,6 +18,7 @@ use embedded_io::asynch::Write;
 use static_cell::StaticCell;
 use {defmt_rtt as _, panic_probe as _};
 
+#[macro_export]
```
Comment (Dirbaio) [inline examples/rpi-pico-w/src/bin/tcp_server.rs:21]: `#[macro_export]` is not needed

File: examples/rpi-pico-w/src/bin/tcp_server.rs [hunk 3]
```diff
@@ -27,7 +28,7 @@ macro_rules! singleton {
 }
 
 #[embassy_executor::task]
-async fn wifi_task(
+pub async fn wifi_task(
```
Comment (Dirbaio) [inline examples/rpi-pico-w/src/bin/tcp_server.rs:31]: `pub` is not needed

**Comment by Dirbaio** [comment]
thanks!

**Pull request merged**